packages/wallet/src/types.ts

```diff
@@ -51,6 +51,25 @@ export type FinalizeTxProps = Omit<TxContext, 'signingContext'> & {
 
 export type HandleInfo = HandleResolution & Asset.AssetInfo;
 
+export interface ScriptAddress {
+  networkId: Cardano.NetworkId;
+  address: Cardano.PaymentAddress;
+  rewardAccount: Cardano.RewardAccount;
+  scripts: {
+    payment: Cardano.NativeScript;
+    stake: Cardano.NativeScript;
+  };
+}
+
+export type WalletAddress = GroupedAddress | ScriptAddress;
+
+export const isKeyHashAddress = (address: WalletAddress): address is GroupedAddress =>
+  Cardano.Address.fromBech32(address.address).asBase()?.getPaymentCredential()?.type === Cardano.CredentialType.KeyHash;
+
+export const isScriptAddress = (address: WalletAddress): address is ScriptAddress =>
+  Cardano.Address.fromBech32(address.address).asBase()?.getPaymentCredential()?.type ===
+  Cardano.CredentialType.ScriptHash;
```

Comment:
That's a lighter implemenatation, could do similar approach for `isKeyHashAddress`
```suggestion
  'scripts' in address
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1030/commits/7134e634d3d1c5437c45f029383bd1213d7756db